src/anemoi/models/preprocessing/monomapper.py

```diff
@@ -0,0 +1,129 @@
+from abc import ABC
+from typing import Optional
+
+import torch
+
+from anemoi.models.data_indices.collection import IndexCollection
+from anemoi.models.preprocessing import BasePreprocessor
+from anemoi.models.preprocessing.mappings import boxcox_converter
+from anemoi.models.preprocessing.mappings import expm1_converter
+from anemoi.models.preprocessing.mappings import inverse_boxcox_converter
+from anemoi.models.preprocessing.mappings import log1p_converter
+from anemoi.models.preprocessing.mappings import sqrt_converter
+from anemoi.models.preprocessing.mappings import square_converter
+
+LOGGER = logging.getLogger(__name__)
+
+
+class Monomapper(BasePreprocessor, ABC):
+    """Remap and convert variables for single variables."""
+
+    supported_methods = {
+        method: [f, inv]
+        for method, f, inv in zip(
+            ["log1p", "sqrt", "boxcox"],
+            [log1p_converter, sqrt_converter, boxcox_converter],
+            [expm1_converter, square_converter, inverse_boxcox_converter],
+        )
+    }
+
+    def __init__(
+        self,
+        config=None,
+        data_indices: Optional[IndexCollection] = None,
+        statistics: Optional[dict] = None,
+    ) -> None:
+        super().__init__(config, data_indices, statistics)
+        self.remappers = []
+        self.backmappers = []
+        for name, method in self.methods.items():
+            method = method or self.default
+            if method == "none":
+                continue
+            elif method in self.supported_methods:
+                self.remappers.append(self.supported_methods[method][0])
+                self.backmappers.append(self.supported_methods[method][1])
+            else:
+                raise ValueError(f"Unknown remapping method for {name}: {method}")
+        self._create_remapping_indices(statistics)
+        self._validate_indices()
+
+    def _validate_indices(self):
+        assert len(self.index_training) == len(self.index_inference) <= len(self.remappers), (
+            f"Error creating conversion indices {len(self.index_training)}, "
+            f"{len(self.index_inference)}, {len(self.remappers)}"
+        )
+
+    def _create_remapping_indices(
+        self,
+        statistics=None,
+    ):
+        """Create the parameter indices for remapping."""
+        # list for training and inference mode as position of parameters can change
+        name_to_index_training = self.data_indices.data.input.name_to_index
+        name_to_index_inference = self.data_indices.model.input.name_to_index
+        self.num_training_vars = len(name_to_index_training)
+        self.num_inference_vars = len(name_to_index_inference)
+
+        (
+            self.index_training,
+            self.index_inference,
+        ) = ([], [])
+
+        # Create parameter indices for remapping variables
+        for name in name_to_index_training:
+            method = self.methods.get(name, self.default)
+            if method == "none":
```

Comment:
see above, a dummy method for 'none' + relying on KeyError would simplify the code here, avoiding nested `ifs`